src/Query/QueryBuilder.php

```diff
@@ -1259,6 +1267,34 @@ public function resetOrderBy(): self
         return $this;
     }
 
+    /**
+     * Adds a CTE to a select query.
+     *
+     * @param string $name         The name of the CTE.
+     * @param self   $queryBuilder The query builder to use within the CTE.
+     *
+     * @return $this This QueryBuilder instance.
+     */
+    public function with(string $name, self $queryBuilder): self
+    {
+        if (! $this->connection->getDatabasePlatform()->supportsCTEs()) {
+            throw new QueryException('CTEs are not supported by this database platform.');
```

Comment:
Should it throw an exception ? It seems that it is supported for all platforms except for mysql 5.7